File: backend/app/schemas/resumen.py
from pydantic import BaseModel
from typing import Optional, List


class DetalleOrigenVenta(BaseModel):
    transacciones: int
    pares_vendidos: int
    ingresos: int
    costos_directos: int
    utilidad_bruta: int


class DesgloseOrigen(BaseModel):
    encargos: DetalleOrigenVenta
    pos: DetalleOrigenVenta


class MetricasFinancieras(BaseModel):
    ingresos_totales: int
    costos_directos: int
    utilidad_bruta: int
    margen_bruto: float
    gastos_operativos: int
    utilidad_neta: int
    margen_neto: float
    transacciones_totales: int
    pares_vendidos: int
    ticket_promedio: int
    desglose_origen: DesgloseOrigen


class DetalleEstadoOperativo(BaseModel):
    cantidad: int
    valor_comprometido: int
    abonos_recibidos: int
    saldo_por_cobrar: int
    utilidad_proyectada: int
    con_costo: int


class PanelOperativo(BaseModel):
    pendientes: DetalleEstadoOperativo
    despachados: DetalleEstadoOperativo
    en_local: DetalleEstadoOperativo
    total_activos: int
    abonos_recibidos_activos: int
    saldo_por_cobrar_activos: int
    utilidad_proyectada_total: int
    cobertura_costos_activos: str


class ItemFlujoFinanciero(BaseModel):
    etiqueta: str
    valor: int
    tipo: str  # 'ingreso' | 'costo' | 'utilidad_bruta' | 'gasto' | 'utilidad_neta'


class ItemGastoCategoria(BaseModel):
    categoria: str
    total: int
    porcentaje: float


class GraficosResumen(BaseModel):
    flujo_financiero: List[ItemFlujoFinanciero]
    gastos_por_categoria: List[ItemGastoCategoria]


class FiltrosResumen(BaseModel):
    periodo: str
    fecha_desde: Optional[str]
    fecha_hasta: Optional[str]


class ResumenFinancieroResponse(BaseModel):
    filtros: FiltrosResumen
    financiero: MetricasFinancieras
    operacional_en_vivo: PanelOperativo
    graficos: GraficosResumen

File: frontend/src/services/resumen.ts
const API = import.meta.env.VITE_API_URL ?? "";

function authHeaders() {
  const token = localStorage.getItem("token");
  return {
    "Content-Type": "application/json",
    ...(token ? { Authorization: `Bearer ${token}` } : {}),
  };
}

export interface DetalleOrigenVenta {
  transacciones: number;
  pares_vendidos: number;
  ingresos: number;
  costos_directos: number;
  utilidad_bruta: number;
}

export interface DesgloseOrigen {
  encargos: DetalleOrigenVenta;
  pos: DetalleOrigenVenta;
}

export interface MetricasFinancieras {
  ingresos_totales: number;
  costos_directos: number;
  utilidad_bruta: number;
  margen_bruto: number;
  gastos_operativos: number;
  utilidad_neta: number;
  margen_neto: number;
  transacciones_totales: number;
  pares_vendidos: number;
  ticket_promedio: number;
  desglose_origen: DesgloseOrigen;
}

export interface DetalleEstadoOperativo {
  cantidad: number;
  valor_comprometido: number;
  abonos_recibidos: number;
  saldo_por_cobrar: number;
  utilidad_proyectada: number;
  con_costo: number;
}

export interface PanelOperativo {
  pendientes: DetalleEstadoOperativo;
  despachados: DetalleEstadoOperativo;
  en_local: DetalleEstadoOperativo;
  total_activos: number;
  abonos_recibidos_activos: number;
  saldo_por_cobrar_activos: number;
  utilidad_proyectada_total: number;
  cobertura_costos_activos: string;
}

export interface ItemFlujoFinanciero {
  etiqueta: string;
  valor: number;
  tipo: string;
}

export interface ItemGastoCategoria {
  categoria: string;
  total: number;
  porcentaje: number;
}

export interface GraficosResumen {
  flujo_financiero: ItemFlujoFinanciero[];
  gastos_por_categoria: ItemGastoCategoria[];
}

export interface FiltrosResumen {
  periodo: string;
  fecha_desde: string | null;
  fecha_hasta: string | null;
}

export interface ResumenFinancieroResponse {
  filtros: FiltrosResumen;
  financiero: MetricasFinancieras;
  operacional_en_vivo: PanelOperativo;
  graficos: GraficosResumen;
}

export async function getResumenFinanciero(
  periodo: string = "mes",
  fechaDesde?: string,
  fechaHasta?: string
): Promise<ResumenFinancieroResponse> {
  const params = new URLSearchParams();
  params.set("periodo", periodo);
  if (fechaDesde) params.set("fecha_desde", fechaDesde);
  if (fechaHasta) params.set("fecha_hasta", fechaHasta);

  const res = await fetch(`${API}/resumen/financiero?${params.toString()}`, {
    headers: authHeaders(),
  });

  if (!res.ok) {
    const errorData = await res.json().catch(() => ({}));
    throw new Error(errorData?.detail || `Error ${res.status} al cargar el resumen`);
  }

  return res.json() as Promise<ResumenFinancieroResponse>;
}
